core/src/main/java/org/infinispan/CacheStream.java

```diff
@@ -190,10 +193,12 @@
 
    /**
     * {@inheritDoc}
-    * <p>This method has special usage when used with a distributed cache backing this set.  This operation will act
-    * as an intermediate iterator operation requiring data be brought locally for proper behavior.  This is
-    * described in more detail in the {@link CacheStream} documentation</p>
-    * <p>This intermediate iterator operation will be performed locally only requiring all elements to be in memory</p>
+    * <p>This method is never invoked on a remote node when used with a distributed cache backing this stream.  This
```

Comment:
The method itself can only be invoked by the user on the local node :)

I'd suggest "This operation is performed entirely on the local node", but then you already say the same thing below.